File: src/main/java/sawyern/cookiebot/repository/LootboxTokenRepository.java
package sawyern.cookiebot.repository;

import org.springframework.data.repository.CrudRepository;
import org.springframework.stereotype.Repository;
import sawyern.cookiebot.models.entity.Account;
import sawyern.cookiebot.models.entity.LootboxToken;

import java.util.List;

@Repository
public interface LootboxTokenRepository extends CrudRepository<LootboxToken, String> {
    List<LootboxToken> findByAccount(Account account);
}

File: src/main/kotlin/sawyern/cookiebot/repository/LootboxTokenRepository.kt
package sawyern.cookiebot.repository

import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import sawyern.cookiebot.models.LootboxToken
import sawyern.cookiebot.models.Season

@Repository
interface LootboxTokenRepository: JpaRepository<LootboxToken, String> {
    fun findByAccountDiscordId(discordId: String): Collection<LootboxToken>
    fun findByAccountDiscordIdAndSeason(discordId: String, season: Season): Collection<LootboxToken>
}